controllers/license/konglicense_controller.go

```diff
@@ -209,19 +284,29 @@ func (r *KongV1Alpha1KongLicenseReconciler) Reconcile(ctx context.Context, req c
 	return ctrl.Result{}, nil
 }
 
+type License struct {
+	License kong.License
+	IsValid mo.Option[bool]
```

Comment:
Would we change the interface of license getter used in translator?
Also, please include some comments to introduce why do we use `mo.Option[bool]` to represent whether the license is valid.

Reply:
I added more explanation. For now, I'm not doing it, because `Agent` from `internal/license/agent.go` uses the must-implement the same interface and I don't want to rework it in this PR.